packages/smooth_app/lib/pages/product_list_user_dialog_helper.dart

```diff
@@ -200,99 +200,47 @@ class ProductListUserDialogHelper {
-      selectedLists = await LoadingDialog.run<List<String>>(
-        context: context,
-        future: daoProductList.getUserLists(withBarcode: barcodes.first),
-      );
-    }
+    final List<String> selectedLists = await LoadingDialog.run<List<String>>(
+          context: context,
+          future: daoProductList.getUserLists(
+            withBarcodes: barcodes.toList(growable: false),
+          ),
+        ) ??
+        <String>[];
 
     return showDialog<bool?>(
       context: context,
-      builder: (BuildContext context) => _UserListsDialogContent(
-        daoProductList: daoProductList,
-        allLists: lists.toSet(),
-        selectedLists: selectedLists?.toSet() ?? <String>{},
-        barcodes: barcodes,
-      ),
-    );
-  }
-}
-
-/// Dialog content to add/remove products from users' lists
-class _UserListsDialogContent extends StatefulWidget {
-  const _UserListsDialogContent({
-    required this.barcodes,
-    required this.allLists,
-    this.selectedLists = const <String>{},
-    required this.daoProductList,
-    Key? key,
-  }) : super(key: key);
-
-  final Set<String> barcodes;
-  final Set<String> allLists;
-  final Set<String> selectedLists;
-  final DaoProductList daoProductList;
-
-  @override
-  State<_UserListsDialogContent> createState() =>
-      _UserListsDialogContentState();
-}
-
-class _UserListsDialogContentState extends State<_UserListsDialogContent> {
-  // To keep track of the changes
-  // The products should be added from these lists
-  final Set<String> _addTo = <String>{};
-  // The products should be removed from these lists
-  final Set<String> _removeFrom = <String>{};
-
-  @override
-  Widget build(BuildContext context) {
-    final AppLocalizations appLocalizations = AppLocalizations.of(context);
-
-    return SmoothAlertDialog(
-      close: true,
-      title: appLocalizations.user_list_add_product,
-      body: _UserLists(
-        lists: widget.allLists,
-        selectedLists: widget.selectedLists,
-        onListSelected: (String list) {
-          _addTo.add(list);
-          _removeFrom.removeWhere((String e) => e == list);
-        },
-        onListUnselected: (String list) {
-          _removeFrom.add(list);
-          _addTo.removeWhere((String e) => e == list);
-        },
-      ),
-      negativeAction: _cancelButton(appLocalizations, context),
-      positiveAction: SmoothActionButton(
-        onPressed: () async {
-          for (final String name in _removeFrom) {
-            await widget.daoProductList.set(
-              ProductList.user(name),
-              // Removal only works for single product addition
-              widget.barcodes.first,
-              false,
-            );
-          }
-
-          for (final String name in _addTo) {
-            await widget.daoProductList
-                .bulkInsert(ProductList.user(name), widget.barcodes.toList());
-          }
+      builder: (BuildContext context) => _UserLists(
+        lists: lists.toSet(),
```

Comment:
You may loose the order, mayn't you? (for the record that's my first "mayn't you?")

Reply:
https://api.dart.dev/stable/2.18.1/dart-core/Set/toSet.html
> Creates a [Set](https://api.dart.dev/stable/2.18.1/dart-core/Set-class.html) with the same elements and behavior as this Set.
> 
> The returned set behaves the same as this set with regard to adding and removing elements. It initially contains the same elements. If this set specifies an ordering of the elements, the returned set will have the same order.

Just double checked your may :D
Looks like the order is kept and this wouldn't be a problem would it? The lists in the list screen are also ordered alphabetically